### DISCUSSION TITLE
GPU support on MacOS with Ollama?

### DISCUSSION THREAD

**Discussion by nickdavis** [open-webui/open-webui#1796] [General]
Hey, sorry I'm confused. I'm on a Mac M3 Max.

I know Ollama has GPU support but if I'm running OpenWebUI on top of Ollama does that mean I'm no longer getting GPU support? 

Asking because of this section in the FAQ::

> **Q: Is GPU support available in Docker?**
> A: GPU support in Docker is available but varies depending on the platform. Officially, GPU support is provided in Docker for Windows and Docker Engine on Linux. Other platforms, such as Docker Desktop for Linux and MacOS, do not currently offer GPU support. This limitation is important to consider for applications requiring GPU acceleration. For the best experience and to utilize GPU capabilities, we recommend using Docker on platforms that officially support GPU integration.

https://docs.openwebui.com/faq#q-is-gpu-support-available-in-docker

Thanks!

**Comment by cheahjs** [comment]
> if I'm running OpenWebUI on top of Ollama does that mean I'm no longer getting GPU support?

Only if you're running Ollama inside of Docker instead of the standard Mac install.

Otherwise, running only Open WebUI inside of Docker on Mac means that only the bits that Open WebUI is responsible for such as running the embedding models for RAG will not be GPU accelerated.

**Reply by nickdavis** [reply]
Ah, thanks, I hoped that was the answer, but wanted to check! That is indeed my setup.

One other 'stupid' question that occurs to me now, if that's OK, regarding this:

> 🤖 **Multiple Model Support**: Seamlessly switch between different chat models for diverse interactions.

This is definitely a cool feature (and working absolutely fine for me from a performance perspective so far) but just curious, when I have different chats going with different models is Ollama stopping and starting each model each time in the background when I switch between the chats? If not, presumably my machine could eventually get overloaded running so many models at once, and therefore I should think about stopping some models myself?

Thanks!
